subs/pantry/src/Pantry/Storage.hs

```diff
@@ -243,39 +258,61 @@ withStorage
 withStorage action =
   flip SQLite.withStorage_ action =<< view (P.pantryConfigL.to P.pcStorage)
 
+
+rdbmsAwareQuery
+  :: MonadIO m
+  => ReaderT SqlBackend m a
+  -> ReaderT SqlBackend m a
+  -> ReaderT SqlBackend m a
+rdbmsAwareQuery postgresQuery sqliteQuery = do
+  rdbms <- connRDBMS <$> ask
+  case rdbms of
+    "postgresql" -> postgresQuery
+    "sqlite" -> sqliteQuery
+    _ -> error $ "rdbmsAwareQuery: unsupported rdbms '" ++ T.unpack rdbms ++ "'"
+
+
+getPackageNameById
+  :: MonadIO m
+  => PackageNameId
+  -> ReaderT SqlBackend m (Maybe P.PackageName)
+getPackageNameById = fmap (unPackageNameP . packageNameName <$>) . get
+
+
 getPackageNameId
-  :: (HasPantryConfig env, HasLogFunc env)
+  :: MonadIO m
   => P.PackageName
-  -> ReaderT SqlBackend (RIO env) PackageNameId
+  -> ReaderT SqlBackend m PackageNameId
 getPackageNameId = fmap (either entityKey id) . insertBy . PackageName . PackageNameP
 
 getVersionId
-  :: (HasPantryConfig env, HasLogFunc env)
+  :: MonadIO m
   => P.Version
-  -> ReaderT SqlBackend (RIO env) VersionId
+  -> ReaderT SqlBackend m VersionId
 getVersionId = fmap (either entityKey id) . insertBy . Version . VersionP
 
-getFilePathId
-  :: (HasPantryConfig env, HasLogFunc env)
-  => SafeFilePath
-  -> ReaderT SqlBackend (RIO env) FilePathId
-getFilePathId = fmap (either entityKey id) . insertBy . FilePath
-
 storeBlob
-  :: (HasPantryConfig env, HasLogFunc env)
+  :: MonadIO m
   => ByteString
-  -> ReaderT SqlBackend (RIO env) (BlobId, BlobKey)
+  -> ReaderT SqlBackend m (BlobId, BlobKey)
```

Comment:
Are these `RIO`-removing changes necessary for Stackage Server, or a cleanup?

Reply:
Some of theses are necessary for Stackage Server others are just cleanup for consistency. RIO doesn't compose very well with Yesod, at least for now, right? So I use RIO inside stackage-server-cron, but everywhere else functions that query database still have to be `ReaderT SqlBackend m`.

I think it is an improvement, since `RIO` wrapper is really unnecessary, especially with `HasPantryConfig env` and `HasLogFunc env` constraints. Also if I remember correctly your suggestion was to try to keep everything in a most general form. Which make this suggestion to work nicely in stackage-server

Reply:
> RIO doesn't compose very well with Yesod, at least for now, right?

Well, sort of. But it seems like all of this can be addressed with a single helper wrapper function in stackage-server. And since we already have such a wrapper function to calling into the database, I would have thought it wouldn't have a big overhead.

> Also if I remember correctly your suggestion was to try to keep everything in a most general form.

Not in general, no. Where there's a strong reason to support non-RIO monads I go in that direction (like the `unliftio` package). But sticking to `RIO` makes error messages much easier to follow, so I prefer it when it won't cause trouble.

Reply:
All switched `ReaderT SqlBackend m` -> `ReaderT SqlBackend (RIO env)`